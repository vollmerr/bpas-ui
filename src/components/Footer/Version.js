import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import theme from '../theme';

const Wrapper = styled.div`
  flex: 3;
  text-align: right;
  padding: ${theme.size.xs}px ${theme.size.sm}px;
  white-space: nowrap;
  color: ${theme.color.primaryInverse};
`;

function Version() {
  return (
    <Wrapper>Version {process.env.VERSION}</Wrapper>
  );
}

export default Version;
