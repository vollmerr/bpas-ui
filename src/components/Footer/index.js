import React from 'react';
import styled from 'styled-components';
import theme from '../../theme';

const Wrapper = styled.footer`

`;

/** Renders the title bars logo  */
function Footer() {
  return (
    <Wrapper>TODO</Wrapper>
  );
}

export default Footer;
