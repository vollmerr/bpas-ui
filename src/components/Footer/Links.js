import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import theme from '../theme';

import Version from './Version';

const urls = [
  {name: 'Conditions of Use', to: 'http://www.ca.gov/Use'},
  {name: 'Privacy Policy', to: 'http://www.ca.gov/Privacy'},
  {name: 'Accessibility', to: 'http://www.ca.gov/Accessibility'},
  {name: 'Contact Us', to: '/help'},
];

const Wrapper = styled.div`
  display: flex;
  flex: 6;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: center;
  margin: 0;
  padding: 0;
`;

const StyledLink = styled(Link)`
  color: ${theme.color.primaryInverse};
  white-space: nowrap;
  min-width: 150px;
  display: flex;
  transition: all 0.25s;

  &:hover,
  &:focus {
    color: ${theme.color.second};
  }
`;

const StyledAnchor = StyledLink.withComponent('a');

function link(url) {
  if(url.to.match(/^http/)) {
    return <StyledAnchor href={url.to} key={url.name}>{url.name}</StyledAnchor>;
  } else {
    return <StyledLink to={url.to} key={url.name}>{url.name}</StyledLink>;
  }
}

function Links() {
  return (
    <Wrapper>
      {urls.map(url => link(url))}
      <Version />
    </Wrapper>
  );
}

export default Links;
