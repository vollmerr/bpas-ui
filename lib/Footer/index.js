var _templateObject = _taggedTemplateLiteral(['\n  min-height: ', 'px;\n  display: flex;\n  align-items: center;\n  background: ', ';\n'], ['\n  min-height: ', 'px;\n  display: flex;\n  align-items: center;\n  background: ', ';\n']);

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

import React from 'react';
import styled from 'styled-components';
import theme from '../theme';

import Container from '../Container';
import Links from './Links';

var Wrapper = styled.footer(_templateObject, theme.size.footer, theme.color.bold);

/** Renders the title bars logo  */
function Footer() {
  return React.createElement(
    Wrapper,
    null,
    React.createElement(
      Container,
      null,
      React.createElement(Links, null)
    )
  );
}

export default Footer;