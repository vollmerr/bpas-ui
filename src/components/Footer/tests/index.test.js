import React from 'react';
import { shallow } from 'enzyme';
import Footer from '../';

process.env.VERSION = 'test 123';

describe('<Footer />', () => {
  it('renders correctly', () => {
    const tree = shallow(
      <Footer />,
      { context: {history: {}} }
    );
    expect(tree).toMatchSnapshot();
  });
});
