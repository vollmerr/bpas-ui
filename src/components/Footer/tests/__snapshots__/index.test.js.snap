// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders correctly 1`] = `
<styled.footer>
  <Container>
    <Links />
  </Container>
</styled.footer>
`;
